lib/providers/badgeview_provider.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:async';
+import 'package:badgemagic/badge_animation/animation_abstract.dart';
+import 'package:badgemagic/constants.dart';
+import 'package:flutter/material.dart';
+
+class DrawBadgeProvider extends ChangeNotifier {
+  int animationIndex = 0;
+  int animationSpeed = aniBaseSpeed.inMilliseconds;
+  int counter = 0;
+  Timer? timer;
+  //List that contains the state of each cell of the badge for home view
+  List<List<bool>> homeViewGrid =
+      List.generate(11, (i) => List.generate(44, (j) => false));
+
+  //List that contains the state of each cell of the badge for draw view
+  List<List<bool>> drawViewGrid =
+      List.generate(11, (i) => List.generate(44, (j) => false));
+
```

Comment:
This needs to be cleaned. 2 seperate widgets for draw and display seperately

Reply:
this I will clear in seperate PR where I will add separate custom paint class for the drawbadge